### PR TITLE
FN: Update attachment_microsoft_image_lure_qr_code.yml

### PR DESCRIPTION
**Pull request by morriscode**
Adding more robust handling when the QR code url does not contain the recipients email.

### CODE REVIEW

File: detection-rules/attachment_microsoft_image_lure_qr_code.yml [hunk 1]
```diff
@@ -19,12 +19,35 @@ source: |
...
-                .scan.qr.type == "url"
-                // recipient email address is present in the URL, a common tactic used in credential phishing attacks 
-                and any(recipients.to, strings.icontains(..scan.qr.data, .email.email))
+            or (
+              any(file.explode(.),
+                  .scan.qr.type == "url"
+                  // recipient email address is present in the URL, a common tactic used in credential phishing attacks 
+                  and any(recipients.to,
+                          strings.icontains(..scan.qr.data, .email.email)
+                          or (
```
Comment (rw-access) [inline detection-rules/attachment_microsoft_image_lure_qr_code.yml:28]: don't need the parentheses here `x or (y or z)` can be flattened to `x or y or z`

File: detection-rules/attachment_microsoft_image_lure_qr_code.yml [hunk 2]
```diff
@@ -19,12 +19,35 @@ source: |
...
+                                strings.icontains(sender.display_name, .email.domain.sld)
+                            )
+                            
+                            // the recipient local is in the body  
+                            or any(recipients.to,
+                                   strings.icontains(body.current_thread.text, .email.local_part)
+                            )
+                            
+                            // or the body is null 
+                            or length(body.current_thread.text) is null
```
Comment (rw-access) [inline detection-rules/attachment_microsoft_image_lure_qr_code.yml:41]: length is null doesn't really make sense and will only be null if the field is also null
```suggestion
                            or body.current_thread.text is null
```

File: detection-rules/attachment_microsoft_image_lure_qr_code.yml (hunk 1 above)
Comment (morriscode) [reply detection-rules/attachment_microsoft_image_lure_qr_code.yml]: Should be fixed!

**Review by rw-access — APPROVED** [review]

**Pull request merged**